app/src/common/shared/org/mozilla/vrbrowser/utils/UrlUtils.java

```diff
@@ -0,0 +1,160 @@
+
+            return uri.toString();
+        } catch (URISyntaxException e) {
+            // We might be trying to display a user-entered URL (which could plausibly contain errors),
+            // in this case its safe to just return the raw input.
+            // There are also some special cases that URI can't handle, such as "http:" by itself.
+            return url;
+        }
+    }
+
+    public static boolean isPermittedResourceProtocol(@Nullable final String scheme) {
+        return scheme != null && (
+                scheme.startsWith("http") ||
+                        scheme.startsWith("https") ||
+                        scheme.startsWith("file") ||
+                        scheme.startsWith("data") ||
+                        scheme.startsWith("javascript") ||
+                        scheme.startsWith("about"));
+    }
+
+    public static boolean isSupportedProtocol(@Nullable final String scheme) {
+        return scheme != null && (isPermittedResourceProtocol(scheme) || scheme.startsWith("error"));
+    }
+
+    public static boolean isInternalErrorURL(final String url) {
+        return "data:text/html;charset=utf-8;base64,".equals(url);
+    }
+
+    /**
+     * Checks that urls are non-null and are the same aside from a trailing slash.
+     *
+     * @return true if urls are the same except for trailing slash, or if either url is null.
+     */
+    public static boolean urlsMatchExceptForTrailingSlash(final String url1, final String url2) {
+        // This is a hack to catch a NPE in issue #26.
+        if (url1 == null || url2 == null) {
+            return false;
+        }
+        int lengthDifference = url1.length() - url2.length();
+
+        if (lengthDifference == 0) {
+            // The simplest case:
+            return url1.equalsIgnoreCase(url2);
+        } else if (lengthDifference == 1) {
+            // url1 is longer:
+            return url1.charAt(url1.length() - 1) == '/' &&
+                    url1.regionMatches(true, 0, url2, 0, url2.length());
+        } else if (lengthDifference == -1) {
+            return url2.charAt(url2.length() - 1) == '/' &&
+                    url2.regionMatches(true, 0, url1, 0, url1.length());
+        }
+
+        return false;
+    }
+
+    public static String stripCommonSubdomains(@Nullable String host) {
+        if (host == null) {
+            return null;
+        }
+
+        // In contrast to desktop, we also strip mobile subdomains,
+        // since its unlikely users are intentionally typing them
+        int start = 0;
+
+        if (host.startsWith("www.")) {
+            start = 4;
+        } else if (host.startsWith("mobile.")) {
+            start = 7;
+        } else if (host.startsWith("m.")) {
+            start = 2;
+        }
+
+        return host.substring(start);
+    }
+
+    public static boolean isLocalizedContent(@Nullable String url) {
```

Comment:
I don't think this function actually does anything right now. If it is supposed to filter out local content like error pages and the home page, it needs to do a lot more than just watch for `about:blank`

Reply:
Making `isLocalizedContent ()` to check if it is `PRIVATE_BROWSING_URI, HOME_WITHOUT_REGION_ORIGIN, getHomeUri, and about:blank`